WordPress/gutenberg
# Separator: Add means to control height

## Description
* Add customizable height to the separator block

## How has this been tested?
Manually.

#### Testing Instructions
1. Checkout this PR, create a new post and add a separator block
2. Select the separator block and adjust its height via the bottom resize handle
3. Confirm the block still displays correctly and that different colors & block styles can be applied.
4. Adjust the separator block's height via the height control in the sidebar control and confirm the block's height updates
5. Change the height units between `px`, `em` or `rem`. Switching units will reset the height. Adjust it again via the sidebar control and confirm visual height of block adjusts.
6. Save the post and confirm the separator's height on the frontend
7. Repeat steps 1-6 using different themes (several use varying methods of styling the block e.g. bottom borders, background colors and pseudo elements).

#### Native Testing Instructions
1. Checkout this PR, start up native emulation e.g. `npm run native start` and `npm run native ios`
2. Add a separator block to the post
3. Open the block's controls and adjust its height in pixels
4. Switch to other height units and ensure that the control's value is set to the default for that unit and the block's height visually adjusts accordingly.

## Screenshots 
![SeparatorHeightV2](https://user-images.githubusercontent.com/60436221/105989563-086fea80-60ed-11eb-9aad-066ba80cb9b8.gif)

[Previous pixel height only iteration screenshot](https://user-images.githubusercontent.com/60436221/105670260-77eaac00-5f2c-11eb-9130-97cd417764c1.gif)

#### iOS
<img src="https://user-images.githubusercontent.com/60436221/106429441-26a36500-64b6-11eb-9398-f31f8c7476b7.gif" width="300"/>

## Types of changes
New feature.

## Checklist:
